subjects/08-flux/exercise/stores/ContactStore.js

```diff
@@ -37,4 +37,37 @@ AppDispatcher.register(function (payload) {
       contacts: action.contacts
     })
   }
+
+  if(action.type === ActionTypes.CONTACT_WAS_DELETED) {
+    setState({
+      loaded: true,
+      contacts: state.contacts.filter((contact) => contact.id !== payload.action.contact.id)
+    })
+  }
+
+  if(action.type === ActionTypes.CONTACT_DELETION_FAILED) {
+    setState({
+      loaded: true,
+      contacts: state.contacts.map((contact) => {
```

Comment:
It's better to put this sort of thing in a different top-level property of this store - mutating `contacts` makes it much harder for the server to push new contacts to the client without breaking things.